docs/cryptodoc/src/05_06_hss_lms.rst

```diff
@@ -0,0 +1,422 @@
+   | LMS_SHAKE_M32_H10  | SHAKE256/256 | 32  | 10  | 0x0010 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M32_H15  | SHAKE256/256 | 32  | 15  | 0x0011 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M32_H20  | SHAKE256/256 | 32  | 20  | 0x0012 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M32_H25  | SHAKE256/256 | 32  | 25  | 0x0013 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M24_H5   | SHAKE256/192 | 24  | 5   | 0x0014 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M24_H10  | SHAKE256/192 | 24  | 10  | 0x0015 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M24_H15  | SHAKE256/192 | 24  | 15  | 0x0016 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M24_H20  | SHAKE256/192 | 24  | 20  | 0x0017 |
+   +--------------------+--------------+-----+-----+--------+
+   | LMS_SHAKE_M24_H25  | SHAKE256/192 | 24  | 25  | 0x0018 |
+   +--------------------+--------------+-----+-----+--------+
+
+In addition to its LMS parameters, an LMS instance (class ``LMS_Instance``) is
+defined by its identifier ``I`` and the LM-OTS parameters used for all
+contained LM-OTS instances.
+We can create a keypair with a secret key (class ``LMS_Private_Key``) and a
+public key (class ``LMS_Public_Key``) for each LMS instance. The secret
+key contains the value ``SEED`` used for LM-OTS secret key derivation, while the
+public key contains the LMS tree's root node. The public key is derived from the
+secret key in the constructor of ``LMS_Public_Key``.
+
+For creating an LMS signature, Botan offers the method
+``LMS_Private_Key::sign_and_pk_gen``, which signs the message and computes the
+public key associated with the LMS instance according to Section 5.3. and 5.4.
+of [RFC8554]_. For verification of a signature-message pair, Botan provides
+``LMS_PublicKey::verify_signature``, implementing  Algorithm 5 of [RFC8554]_.
+The internal logic to create and reconstruct Merkle trees is implemented in the
+cross-algorithm helper module ``tree_hash``
+(:srcref:`src/lib/utils/tree_hash/tree_hash.h`). The tree's leaves are created using the
+constructs introduced in :ref:`Section LM-OTS <pubkey/hss_lms/lm_ots>`.
+
+.. _pubkey/hss_lms/hss:
+
+HSS
+^^^
+
+Multiple LMS trees build up an HSS/LMS hypertree, where leaf nodes of higher LMS
+trees sign the public keys of lower LMS instances. The following
+parameters define the HSS/LMS hypertree. The parameter ``L`` configures the height
+of the HSS/LMS hypertree, i.e., the number of LMS tree levels in the hypertree.
+As specified in [RFC8554]_, Botan allows ``L`` to be within one and eight.
+An LMS and LM-OTS parameter set pair is defined for each level. Botan allows all
+parameter combinations as long as the hash function
+is always the same in every set at every level.
```

Comment:
Why? Does this follow the philosophy of the RFC? If yes, then better state it here, so that it does not appear as a random choice.

Reply:
Yeah, it is a NIST SP 800-208 recommendation:
>If the HSS instance has more than one level, then the hash function used for the tree at level 0 shall be used for every LMS tree at every other level.

I agree, a reference is missing here!